tools/topology/CMakeLists.txt

```diff
@@ -132,6 +132,8 @@ set(TPLGS
 	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DPLATFORM=tgl\;-DAMP_SSP=1"
 	"sof-tgl-max98357a-rt5682\;sof-adl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DPLATFORM=adl\;-DAMP_SSP=2"
 	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682-pdm1\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DDMIC_DAI_LINK_16k_PDM=STEREO_PDM1\;\;-DPLATFORM=tgl\;-DAMP_SSP=1"
+	"sof-tgl-max98357a-rt5682\;sof-adl-max98390-rt5682\;-DCODEC=MAX98390\;-DFMT=s32le\;-DPLATFORM=adl\;-DAMP_SSP=1"
+	"sof-tgl-max98357a-rt5682\;sof-adl-max98390-ssp2-rt5682-ssp0\;-DCODEC=MAX98390\;-DFMT=s32le\;-DPLATFORM=adl\;-DAMP_SSP=2"
```

Comment:
in your kernel comments you mentioned 3 configurations, I still see 2 here?

Reply:
By config 4 ch tdm slots that support 2 speakers(in slot0/1) and 4 speakers(slot 0/1/2/3) depends on real codec dais numbers. Or should I mentioned 2 hw boards in case of confusion?
So 
board1: SSP1 x 2 speakers (sof-adl-max98390-rt5682.tplg)
board2: SSP1 4 speakers (sof-adl-max98390-rt5682.tplg)
plus
board3: SSP2 x 2 speakers (with sof-adl-max98390-ssp2-rt5682-ssp0.tplg)

Reply:
Not following @macchian 

> By config 4 ch tdm slots that support 2 speakers(in slot0/1) and 4 speakers(slot 0/1/2/3) depends on real codec dais numbers. Or should I mentioned 2 hw boards in case of confusion?
> So
> board1: SSP1 x 2 speakers (sof-adl-max98390-rt5682.tplg)
> board2: SSP1 4 speakers (sof-adl-max98390-rt5682.tplg)

You have the same topology for a different configuration of the SSP1. How would you manage the clock and slot differences with the same topology file? Even if you play in stereo you need 4 slots for capture. Something's missing or I don't get your idea.

> plus
> board3: SSP2 x 2 speakers (with sof-adl-max98390-ssp2-rt5682-ssp0.tplg)

And here you mentioned SSP2 but with an -ssp0 suffix, so it's hard to follow!

Reply:
These parts are relevant to Maxim. 
https://lore.kernel.org/all/20210405143801.29770-1-steves.lee@maximintegrated.com/

Share the 2/4 configuration table in DT from Maxim. 

Dev Name	                          Slave address	slot configuration
MX98390:02 Tweeter Right 	0x38	            maxim,vmon-slot-no = 2
		                                                            maxim,imon-slot-no = 3
MX98390:03 Tweeter Left	0x39	            maxim,vmon-slot-no = 3
		                                                            maxim,imon-slot-no = 2
MX98390:00 - Right	        0x3a	                    maxim,vmon-slot-no = 0
		                                                            maxim,imon-slot-no = 1
MX98390:01 - Left	               0x3b	                    maxim,vmon-slot-no = 1
		                                                            maxim,imon-slot-no = 0

Even if you play in stereo you need 4 slots for capture?
In stereo case, in the same tdm configuration, the echoRef recorded slot 0 and slot1 capture.

board3: SSP2 x 2 speakers (with sof-adl-max98390-ssp2-rt5682-ssp0.tplg)
I thought the -ssp0 suffix is described for rt5682? I follow the name like
"sof-tgl-max98373-rt5682\;sof-tgl-rt5682-ssp0-max98373-ssp2\;-DAMP_SSP=2"

Reply:
what I meant is that we use 2 slots for playback, regardless of how many amplifiers there are. But in the two amp case, we use 2 slots for capture and in the four-amp case we use 4 slots.
That means we need to use two different topologies since the SSP configuration will be different. To be clearer, ONLY the SSP configuration will be different (DAI_CONFIG)

Reply:
@plbossart ,  sorry for confusing. yes, base on this topology here is as you thought, no matters how many amplifiers there 2 slots for playback, 2 slot for capture. 4 slots for playback , 4 slots for capture.

Reply:
Even if you play in stereo you need 4 slots for capture.
My bad to don't answer this correctly. Yes, in stereo playback, it needs 4 slots for capture. So 

> @plbossart , sorry for confusing. yes, base on this topology here is as you thought, no matters how many amplifiers there 2 slots for playback, 2 slot for capture. 4 slots for playback , 4 slots for capture.

@plbossart my bad... please ignore my unclear comments. I answer it incorrectly. No matters how many amplifiers here 4 slots for playback and 4 slots for capture. That's the SSP1 board design.

Reply:
@macchian would be good to add these details about channel mappings due to board design in the commit message so everyone can follow.

Reply:
Hi @plbossart , @lgirdwood , do you have further comments, good to go?

Reply:
Just a little more in the commit due to the complexity its best to state it like 
```
Board X with Codec Y uses 4 channels mapped as follows for Z (playback, capture ...)
Chan 0 = Left
Chan 1 = Right
Chan 2 = Tweeter Left
Chan 3 = Tweeter Right

And so on
```